frontend/catalyst/pennylane_extensions.py

```diff
@@ -2015,22 +2032,31 @@ def circuit():
     if len(wires) != 1:
         raise TypeError(f"One classical argument (a wire) is expected, got {wires}")
 
-    in_classical_tracers = wires
+    # Copy, so wires remain unmodified
+    in_classical_tracers = wires.copy()
 
-    # Check the postselect value. If given, add it to the classical tracers list
-    if postselect is not None:
-        if postselect not in [0, 1]:
-            raise TypeError(f"postselect must be '0' or '1', got {postselect}")
-        in_classical_tracers.append(postselect)
+    if postselect is not None and postselect not in [0, 1]:
+        raise TypeError(f"postselect must be '0' or '1', got {postselect}")
+    in_classical_tracers.append(postselect)
```

Comment:
I guess this is from a previous PR, but it's interesting that the value is processed as a "tracer" when we restrict the values of postselect to be constants.

Reply:
Yes, this comes from the post-selection PR. I didn't see any other way to pass the postselect value but through the `in_classical_tracers` var. Is there another way?

Reply:
Good question, I guess you would need to re-add a custom constructor to the MidCircuitMeasure class which was removed at some point. Then you could store the value in a `postselect` attribute or similar.